Added test and improved task ordering

rgozim opened the pull request:
See commit message

rgozim committed:
Working on integration tests
Added string convenience methods to tasks
Set spock framework version to 1.2-groovy-2.5
Added tests for OmeXmlLoader and DslSlingleFilesTask
Added default outputPath for DSLPlugin
Removed plugin version from tests
Added multi file task test
Moved logic for task ordering from dslpluginbase to dslplugin
Removed final keyword from tests
Used dslpluginbase task prefix
Fixed gstring syntax error
Fixed task dependency ordering

rgozim committed:
Reinstated support for mapping properties types
(more work required, if this is to be configurable via gradle)

jburel commented on dsl/src/main/java/ome/dsl/JavaToDbType.java:
```diff
@@ -0,0 +1,39 @@
+
+/**
+ * Be mindful that this is a Singleton class.
+ * It's purpose is to lookup a map a small number of Java types (such as string[]) to
```
Its

jburel commented on dsl/src/main/java/ome/dsl/velocity/Generator.java:
```diff
@@ -86,41 +87,18 @@ protected Generator(Builder builder) {
-        InputStream in = getTemplateStream(template);
-        OutputStream os = getOutputStream(output);
+    void parseTemplate(VelocityContext vc, File template, File output) throws IOException {
+        InputStream in = FileUtils.openInputStream(template);
```
the streams won't be closed if an error occurs during initialisation

jburel commented:
With this changes we are back to the state on develop
i.e.
```
@javax.persistence.Column(columnDefinition="", nullable=true, unique=false, name="textValue", updatable=true)
   public java.lang.String getTextValue() {
```

joshmoore commented:
I'm a bit confused why object.vm is a fully new file here, but considering omero-dsl and omero-model need rewriting, I can handle this after this PR.

rgozim committed:
Added some velocity config options

rgozim committed:
Added further config options to velocity and added default option set to DSLPlugin class

rgozim committed:
Small fixup in default properties setting for velocity

jburel commented:
We may also want to set:
```
runtime.conversion.handler = none
space.gobbling = bc
```
as suggested on http://velocity.apache.org/engine/2.0/upgrading.html

jburel commented:
The ``emptycheck`` is set and the ``Shape`` class is correctly generated

rgozim committed:
Removed .ome.xml files

Removed .vm test resources and other dsl integration tests

Removed unused sourceSets from build.gradle

jburel committed:
Fix typo

jburel committed:
make sure streams are closed

jburel committed:
remove space leading to build error

rgozim committed:
Extension fixup

rgozim committed:
Trying to fix task evaluation

rgozim committed:
Merge branch 'develop' of github.com:rgozim/omero-dsl into develop

rgozim committed:
Fixed some small setter bugs in extensions

rgozim committed:
Fixed DSL task ordering for generateXXX tasks

jburel committed:
Add option to set the profile e.g. psql. No default set

jburel committed:
remove ref to default

jburel committed:
refactor to read in Generator

jburel committed:
Clean up

jburel committed:
update doc

jburel committed:
review check

jburel committed:
remove file no longer needed

rgozim committed:
Added some logic in dslbase task to extract .ome.xml files from paths

rgozim committed:
Merge branch 'develop' of github.com:rgozim/omero-dsl into develop

rgozim committed:
Added some logic in DslBaseTask to extract .ome.xml files from paths

rgozim committed:
Added default outputDir for generated content

jburel committed:
fix typo

rgozim committed:
DatabaseTypes changed to File type for gradle task caching

jburel commented:
@rgozim your last commit has broken the generation of omero-model

rgozim committed:
Fixed bugs

rgozim committed:
Added databaseType string as database type selector

rgozim committed:
Moved some convention config of some properties

jburel committed:
revert change, it prevents omero-model to be built correctly

rgozim committed:
Fixed outputDir path issue

jburel committed:
upgrade gradle

joshmoore commented:
Merging in prep for 5.5.0-m3
